go/vt/wrangler/reparent.go

```diff
@@ -519,6 +519,21 @@ func (wr *Wrangler) plannedReparentShardLocked(ctx context.Context, ev *events.R
 		return fmt.Errorf("failed to PopulateReparentJournal on master: %v", err)
 	}
 
+	// After the master is done, we can update the shard record.
+	// This is still necessary because we might have version skew during
+	// upgrades. Some vttablets might be running a newer version of the code
+	// and some might still be running the previous version where tablets
+	// are not responsible for updating the shard record.
+	// We can remove this block of code in 5.0
+	wr.logger.Infof("updating shard record with new master %v", masterElectTabletAlias)
+	if _, err := wr.ts.UpdateShardFields(ctx, keyspace, shard, func(si *topo.ShardInfo) error {
+		si.MasterAlias = masterElectTabletAlias
```

Comment:
If we update the alias without updating the timestamp, the record is in an inconsistent state. This is probably acceptable during the initial upgrade, but we'd want to avoid this race at steady-state after the upgrade is complete. On the other hand, it's not safe to set `time.Now()` here because it might conflict with the tablet's internal concept of its timestamp.

Does this problem go away if we say that you have to update all tablets first, before updating any vtctlds (as recommended in the vitess.io docs)?

Reply:
Actually, that is right. If all vttablets have been upgraded, then vtctld (wrangler) no longer needs to do this update.